src/main/java/com/qouteall/immersive_portals/commands/MyCommandServer.java

```diff
@@ -996,6 +1111,39 @@ private static int placePortalShift(CommandContext<ServerCommandSource> context)
         return 1;
     }
 
+    private static int teleport(Collection<? extends Entity> entities, DimensionType targetDim, Vec3d targetPos) {
+        ServerWorld targetWorld = McHelper.getServer().getWorld(targetDim);
+        
+        int numTeleported = 0;
+
+        for (Entity entity : entities) {
+            if (entity instanceof ServerPlayerEntity) {
+                Global.serverTeleportationManager.invokeTpmeCommand(
+                    (ServerPlayerEntity) entity, targetDim, targetPos
+                );
+            } else {
+                if (targetWorld == entity.world) {
+                    entity.refreshPositionAndAngles(targetPos.x, targetPos.y, targetPos.z, entity.yaw, entity.pitch);
+                    entity.setHeadYaw(entity.yaw);
+                } else {
+                    entity.detach();
```

Comment:
Can be replaced by ServerTeleportationManager.changeEntityDimension

Reply:
Then I'd have to calculate the eye position rather than setting normal position

Reply:
just subtract the eye height